src/content/mod_configuration.h

```diff
@@ -83,10 +84,16 @@ class ModConfiguration
 
 	/**
 	 * Call this function once all mods have been added
+	 *
+	 * @param first_mod First mod to be loaded
+	 * @param last_mod Last mod to be loaded
 	 */
 	void checkConflictsAndDeps();
 
 private:
+	std::optional<std::string> m_first_mod;
+	std::optional<std::string> m_last_mod;
```

Comment:
fwiw I think it's okay to use an empty string as an implicit null, avoid possible confusion between `""` (often not valid) and `std::nullopt`

Reply:
Looks like @appgurueu should answer this.

Reply:
I could have left this as it was tbh, both are fine. Though since I already converted some to optionals, it improves consistency to also use optionals here. Strings are just used as mod "IDs" / "names" here. When I see a `std::string` in the context of a name / ID, I don't know if an empty string is allowed (a comment could help communicate this, but only by convention), and I can easily forget to handle the empty string properly.

`std::optional<std::string>` makes this pretty clear IMO if you know that the `std::string` is conceptually a name (and hence not supposed to be empty), though something like `std::optional<NonEmptyString>` would be better.

If we were to make this even cleaner, we could do `std::optional<ModName>` and `using ModName = std::string;`, and if we were to make this completely idiot-proof we could even have `ModName` be a type that enforces a valid, non-empty mod name (this would be overkill, though).

Reply:
> `std::optional<std::string>` makes this pretty clear IMO if you know that the `std::string` is conceptually a name (and hence not supposed to be empty),

I would have interpreted that exactly opposite: someone went to the effort of adding `std:optional<>`, so there must be something special about `""` that makes it different from a null value. maybe it's valid too?

Reply:
So, should I revert the last commit (std::optional add)?

`fistModSpec` and `lastModSpec` naming is now fixed in the fist commit also.

Reply:
@appgurueu opinion?

Reply:
Seeing as it seems to be considered counterproductive, let me undo the optional-ification of these two strings.